spark3/src/main/java/org/apache/iceberg/spark/source/SparkWrite.java

```diff
@@ -365,6 +365,7 @@ private void commitWithSerializableIsolation(OverwriteFiles overwriteFiles,
 
       Expression conflictDetectionFilter = conflictDetectionFilter();
       overwriteFiles.validateNoConflictingAppends(conflictDetectionFilter);
+      overwriteFiles.validateNoConflictingDeleteFiles(conflictDetectionFilter);
```

Comment:
@rdblue, could you double check this place?

Reply:
The check looks good to me.